File: Javascript/array/binarysearch.js
Arr = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,23,26,34,67,89]

function binarySearch(Arr, key) {
    let low = 0
    let high = Arr.length - 1
    let mid;
    while(low <= high) {
        mid = Math.floor((low + high) / 2)
        if (Arr[mid] == key) {
            return mid
        } else if (Arr[mid] > key) {
            high = mid - 1
        } else {
            low = mid + 1
        }

        if (low > high) {
            return -1
        }
        

    }
}

console.log(binarySearch(Arr, 89))

File: C Programming/search/binarySearch.c
#include<stdio.h>

int binarySearch(int arr[], int low, int high, int key) {
    if(low > high) {
        return -1;
    }
    int mid = (low + high)/2;

    while(low<high) {
        if (arr[mid] == key)
        { 
            return mid;
        
        }
        else if(arr[mid] > key){
            return binarySearch(arr,low,mid-1,key);
        }
        else {
            return binarySearch(arr,mid+1,high,key);
        }
    }
    return -1;

}


int main() {
    int arr[100], n, i;
    printf("Enter the number of elements: ");
    scanf("%d", &n);
    printf("Enter the elements: ");
    for(i=0; i<n; i++) {
        scanf("%d", &arr[i]);
    }
    printf("Enter the element to be searched: ");
    int key;
    scanf("%d", &key);
    int result = binarySearch(arr, 0, n-1, key);
    if(result == -1) {
        printf("Element not found\n");
    } else {
        printf("Element found at index %d\n", result);
    }

    return 0;
}